File: applications/space/client/office/src/core/model/serializerdata.js
var SerializerData = new Object();

SerializerData.ITEM_SEPARATOR = "#";
SerializerData.ITEM_CODE_SEPARATOR = "=";

SerializerData.deserialize = function (data) {
  var result = new Array();

  if (data == null || data == "") return result;

  var itemsArray = data.split(SerializerData.ITEM_SEPARATOR);
  for (var i = 0; i < itemsArray.length; i++) {
    var itemArray = itemsArray[i].split(SerializerData.ITEM_CODE_SEPARATOR);
    if (itemArray.length == 0)
      continue;
    result[itemArray[0]] = (itemArray.length > 1) ? itemArray[1] : "";
  }

  return result;
};

SerializerData.deserializeSet = function (data) {
  var result = new Array();

  if (data == null || data == "") return result;

  var itemsArray = data.split(SerializerData.ITEM_SEPARATOR);
  for (var i = 0; i < itemsArray.length; i++)
    result.push(itemsArray[i]);

  return result;
};

SerializerData.serialize = function (data) {
  var result = "";

  if (data == null) return "";

  for (var key in data) {
    if (isFunction(data[key])) continue;
    var value = data[key];
    result += key + SerializerData.ITEM_CODE_SEPARATOR + value + SerializerData.ITEM_SEPARATOR;
  }

  if (result.length > 0)
    result = result.substring(0, result.length - 1);

  return result;
};

SerializerData.serializeSet = function (data) {
  var result = "";

  if (data == null) return "";

  for (var key in data) {
    if (isFunction(data[key])) continue;
    result += data[key] + SerializerData.ITEM_SEPARATOR;
  }

  if (result.length > 0)
    result = result.substring(0, result.length - 1);

  return result;
};

File: applications/space/server/src/org/monet/space/kernel/model/SerializerData.java
package org.monet.space.kernel.model;

import java.util.*;
import java.util.Map.Entry;

public abstract class SerializerData {

	private static final String ITEM_SEPARATOR = "#";
	private static final String ITEM_CODE_SEPARATOR = "=";

	public static <T extends Map<String, String>> T deserialize(String data) {
		LinkedHashMap<String, String> result = new LinkedHashMap<String, String>();

		if (data == null || data.isEmpty())
			return (T) result;

		String[] itemsArray = data.split(SerializerData.ITEM_SEPARATOR);
		for (int i = 0; i < itemsArray.length; i++) {
			String[] itemArray = itemsArray[i].split(SerializerData.ITEM_CODE_SEPARATOR);
			if (itemArray.length == 0)
				continue;
			result.put(itemArray[0], ((itemArray.length > 1) ? itemArray[1] : ""));
		}

		return (T) result;
	}

	public static <T extends Set<String>> T deserializeSet(String data) {
		LinkedHashSet<String> result = new LinkedHashSet<String>();

		if (data == null || data.isEmpty())
			return (T) result;

		String[] itemsArray = data.split(SerializerData.ITEM_SEPARATOR);
		for (int i = 0; i < itemsArray.length; i++)
			result.add(itemsArray[i]);

		return (T) result;
	}

	public static <S> String serialize(Map<String, S> data) {
		String result = "";

		if (data == null)
			return "";

		for (Entry<String, S> entry : data.entrySet())
			result += entry.getKey() + SerializerData.ITEM_CODE_SEPARATOR + entry.getValue() + SerializerData.ITEM_SEPARATOR;

		if (result.length() > 0)
			result = result.substring(0, result.length() - 1);

		return result;
	}

	public static String serializeSet(Set<String> data) {
		String result = "";

		if (data == null)
			return "";

		for (String value : data)
			result += value + SerializerData.ITEM_SEPARATOR;

		if (result.length() > 0)
			result = result.substring(0, result.length() - 1);

		return result;
	}

}
